src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -400,6 +401,111 @@ internal static SqlConnectionAttestationProtocol ConvertToAttestationProtocol(st
 
         #endregion
 
+        #region <<IPAddressPreference Utility>>
+        /// <summary>
+        /// IP Address Preference.
+        /// </summary>
+        private readonly static Dictionary<string, SqlConnectionIPAddressPreference> s_preferenceNames = new(StringComparer.InvariantCultureIgnoreCase);
+
+        static DbConnectionStringBuilderUtil()
+        {
+            foreach (SqlConnectionIPAddressPreference item in Enum.GetValues(typeof(SqlConnectionIPAddressPreference)))
+            {
+                s_preferenceNames.Add(item.ToString(), item);
+            }
+        }
+
+        /// <summary>
+        ///  Convert a string value to the corresponding IPAddressPreference.
+        /// </summary>
+        /// <param name="value">The string representation of the enumeration name to convert.</param>
+        /// <param name="result">When this method returns, `result` contains an object of type `SqlConnectionIPAddressPreference` whose value is represented by `value` if the operation succeeds. 
+        /// If the parse operation fails, `result` contains the default value of the `SqlConnectionIPAddressPreference` type.</param>
+        /// <returns>`true` if the value parameter was converted successfully; otherwise, `false`.</returns>
+        internal static bool TryConvertToIPAddressPreference(string value, out SqlConnectionIPAddressPreference result)
+        {
+            if (!s_preferenceNames.TryGetValue(value, out result))
+            {
+                result = DbConnectionStringDefaults.IPAddressPreference;
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Verifies if the `value` is defined in the expected Enum.
+        /// </summary>
+        internal static bool IsValidIPAddressPreference(SqlConnectionIPAddressPreference value)
+            => value == SqlConnectionIPAddressPreference.IPv4First
+                || value == SqlConnectionIPAddressPreference.IPv6First
+                || value == SqlConnectionIPAddressPreference.UsePlatformDefault;
+
+        internal static string IPAddressPreferenceToString(SqlConnectionIPAddressPreference value)
+            => Enum.GetName(typeof(SqlConnectionIPAddressPreference), value);
+
+        internal static SqlConnectionIPAddressPreference ConvertToIPAddressPreference(string keyword, object value)
+        {
+            if (value is null)
+            {
+                return DbConnectionStringDefaults.IPAddressPreference;  // IPv4First
+            }
+
+            string sValue = (value as string);
+            if (sValue is not null)
```

Comment:
Can this be `if (value is string sValue)` ? the `is` operator does the type and null checks so it's known to be a non-null string inside the if block if the condition returns true.